util/file_reader_writer.h

```diff
@@ -45,6 +47,8 @@ class SequentialFileReader {
   Status Skip(uint64_t n);
 
   SequentialFile* file() { return file_.get(); }
+
+  bool UseDirectIO() const { return file_->UseDirectIO(); }
```

Comment:
I think I would do that after landing both write and read to make the logic clear.